pkg/codegen/internal/test/sdk_driver.go

```diff
@@ -207,8 +231,21 @@ type SDKCodegenOptions struct {
 func TestSDKCodegen(t *testing.T, opts *SDKCodegenOptions) { // revive:disable-line
 	testDir := filepath.Join("..", "internal", "test", "testdata")
 
-	for _, tt := range sdkTests {
-		t.Run(tt.Description, func(t *testing.T) {
+	// Motivation for flagging parallelism: not all tests are
+	// parallel-safe yet (for example, codegen/docs tests fail),
+	// and there are concerns about memory utilizaion in CI. It
+	// can be a nice feature for developing though.
+	parallel := os.Getenv("PULUMI_PARALLEL_SDK_CODEGEN_TESTS") != ""
```

Comment:
YMMV but it helps a lot on my laptop to use that plus `-parallel 4`.

Reply:
I agree that parallelism should be flagged because of memory utilization. If parallelism is known to break certain tests, we should disable it for those tests. That way we can run `PULUMI_PARALLEL_SDK_CODEGEN_TESTS=true go test ./...`.

Reply:
I will take that as follow up.